[bitnami/template] Add scripts and instructions for generating new charts from the template

dan-hill2802 opened the pull request:
### Description of the change

Add a means to automate much of the effort in creating new charts from the template chart

### Benefits

Reduced effort in creating new charts. More consistency in new charts.

### Possible drawbacks

Feature should work on Linux and Mac operating systems, but not Windows. Windows users should use WSL (Windows Subsystem Linux) or a VM.

### Applicable issues

- Potentially fixes #20350 

### Additional information

Requires `make` and `docker`

### Checklist

- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

andresbono commented on Makefile:
```diff
@@ -0,0 +1,34 @@
+.PHONY: new_chart
+.ONESHELL:
+new_chart:  ## Creates new chart from template
+	@command -v docker >/dev/null || { echo "please install 'docker' and enusre on PATH"; exit 1; }
```
```suggestion
	@command -v docker >/dev/null || { echo "please install 'docker' and ensure on PATH"; exit 1; }
```

andresbono commented on Makefile:
```diff
@@ -0,0 +1,34 @@
+.PHONY: render_template
+.ONESHELL:
+render_template:  ## Render template created with 'new_template' and values from its placeholders.yaml
+	@command -v docker >/dev/null || { echo "please install 'docker' and enusre on PATH"; exit 1; }
```
```suggestion
	@command -v docker >/dev/null || { echo "please install 'docker' and ensure on PATH"; exit 1; }
```

andresbono commented on Makefile:
```diff
@@ -0,0 +1,34 @@
+.PHONY: recreate_chart
+.ONESHELL:
+recreate_chart:  ## Removes existing chart, then recreates chart from template then renders with previous placeholders.yaml. Warning: all other changes will be lost.
+	@command -v docker >/dev/null || { echo "please install 'docker' and enusre on PATH"; exit 1; }
```
```suggestion
	@command -v docker >/dev/null || { echo "please install 'docker' and ensure on PATH"; exit 1; }
```

andresbono commented on README.md:
```diff
@@ -66,6 +67,33 @@ Useful Helm Client Commands:
 
+## Creating a new chart
+
+To make a chart that follows the same structure/patterns that the rest of the Bitnami charts, the basic scaffolding is provided in the [`template` directory](https://github.com/bitnami/charts/tree/main/template).
```
Why don't you use a relative reference for `./template` as you are doing for other links? Does it make sense?

dan-hill2802 replied:
I copied the link from CONTRIBUTING.md, so kept it as it was. I later updated the linter to allow relative links. I have now made this link relative too.

andresbono commented on README.md:
```diff
@@ -66,6 +67,33 @@ Useful Helm Client Commands:
+
+This will replace all the placeholders throughout the chart with your values. You will still likely need to modify the chart further to meet your needs.
+
+The new chart can then me moved from `/draft/<chart-name>` to your own repo/location. If you wish to add the chart to this Bitnami repository, then refer to [Adding a new chart to the repository](./CONTRIBUTING.md#adding-a-new-chart-to-the-repository)
```
```suggestion
The new chart can then be moved from `/draft/<chart-name>` to your own repo/location. If you wish to add the chart to this Bitnami repository, then refer to [Adding a new chart to the repository](./CONTRIBUTING.md#adding-a-new-chart-to-the-repository)
```

andresbono reviewed (commented):
Thank you for your contribution! I have just added suggestions for minor typo fixes to be addressed. Could you please check them?

We will give a more thorough review on a second round of review.

Thank you so much again!

dan-hill2802 commented:
> Thank you for your contribution! I have just added suggestions for minor typo fixes to be addressed. Could you please check them?
> 
> We will give a more thorough review on a second round of review.
> 
> Thank you so much again!

All issues now addressed.

andresbono commented:
Thank you for addressing the minor suggestions, @dan-hill2802. Having another look to the PR, I wonder if 3 shell scripts plus a `Makefile` isn't too much for the relatively simple task we want to accomplish with them. I think it would suffice with a single shell script under `./template/`. The feature is not going to be part of any CI/CD action for now. What do you think?

dan-hill2802 commented:
> Thank you for addressing the minor suggestions, @dan-hill2802. Having another look to the PR, I wonder if 3 shell scripts plus a `Makefile` isn't too much for the relatively simple task we want to accomplish with them. I think it would suffice with a single shell script under `./template/`. The feature is not going to be part of any CI/CD action for now. What do you think?

Hi @andresbono, just to confirm I understand, are you suggesting refactoring the three scripts into 3 separate functions within one script file then executed from make like this: `./template/run.sh new-chart $$chart_name` and `./template/run.sh render-template $$chart_name`? If so, then that makes sense and I can make that change.

andresbono commented:
Hi again @dan-hill2802, after gathering more feedback from others teammates about this feature and a thorough discussion, even simplifying the logic as I'm suggesting, we consider the PR is not going to add much value in the end and we will not merge it.

These are some of the main reasons:

* The template dir is not supposed to be converted into a chart directly for basic scaffolding, (kind of what [helm create](https://helm.sh/docs/helm/helm_create/) does). We can think of it more as a library where you pick the specific templates you need **after** the architecture of the chart is defined.
* It will increase the maintenance load, more logic to keep up-to-date for something that is not going to add much value.

I hope you understand, we really appreciate your efforts on putting this PR together. Do not hesitate to submit new contributions!!

dan-hill2802 commented:
Thanks for letting me know @andresbono. It's a little disappointing, but I understand your points and agree that it can never be an automated process, as creating a chart requires many architectural decisions. The goal was mainly to reduce the effort and chance of error while copying the template and replacing the placeholders.
I will maintain my fork with this feature. If it is deemed useful in the future it should still be able to merge, as I don't anticipate many conflicts.
